packages/mdc-dialog/foundation.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MDCFoundation} from '@material/base';
+import {cssClasses, strings} from './constants';
+
+export default class MDCDialogFoundation extends MDCFoundation {
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  static get strings() {
+    return strings;
+  }
+
+  static get defaultAdapter() {
+    return {
+      setBackgroundAriaHidden: (/* el: Element */) => {},
+      setDialogAriaHidden: (/* el: Element */) => {},
+      hasClass: (/* className: string */) => {},
+      addClass: (/* className: string */) => {},
+      removeClass: (/* className: string */) => {},
+      addScrollLockClass: (/* className: string */) => {},
+      removeScrollLockClass: (/* className: string */) => {},
+      registerInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      deregisterInteractionHandler: (/* evt: string, handler: EventListener */) => {},
+      registerDialogInteractionHandler: (/* evt: string, handler: EventListener */) => {},
```

Comment:
Gotcha. Could we rename these methods to `{de}registerDialogSurfaceInteractionHandler`? It's verbose but IMO it's clearer as to what the method should do since the name is indicative of using `dialog__surface`.